slick/src/main/scala/slick/jdbc/MySQLProfile.scala

```diff
@@ -94,6 +94,12 @@ trait MySQLProfile extends JdbcProfile { profile =>
         else l
       }
     }
+
+    //Reference: https://github.com/slick/slick/issues/1419
+    override def createTableNamer(meta: MTable): TableNamer = new TableNamer(meta){
+      override def schema = meta.name.catalog
+      override def catalog = meta.name.schema 
```

Comment:
```
In MySQL:

    server instance == not identified with catalog, just a set of databases
    database == schema == catalog == a namespace within the server.
    user == named account, who is can connect to server and use (but can not own - no concept of ownership) objects in one or more databases
    to identify any object you need (database name + object name)

```

[http://stackoverflow.com/questions/7942520/relationship-between-catalog-schema-user-and-database-instance](http://stackoverflow.com/questions/7942520/relationship-between-catalog-schema-user-and-database-instance)

I think the current implementation is ok also considering @cvogt 's comment.